### PR TITLE
[WIP] Become plugins

### PR DESCRIPTION
**Pull request by abadger**
##### SUMMARY
become_plugins using the CLIARGS context so we don't have to pass options around everywhere

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
become plugins

##### ADDITIONAL INFORMATION

The main thing yet to do is to get rid of play.Play._set_from_CLI_options and base.FieldAttributeBase._set_options.  Those should be implemented via FieldAttribute defaults instead.

This is built on top of https://github.com/ansible/ansible/pull/50069 which should be merged first.

### CODE REVIEW

File: lib/ansible/plugins/become/pmrun.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      - This plugin ignores the become_user supplied and uses pmrun's own configuration to select the user.
+"""
+
+from ansible.plugins.become import BecomeBase
+
+
+class BecomeModule(BecomeBase):
+
+    name = 'pmrun'
+    prompt = 'Enter UPM user password:'
```
Comment (AlanCoding) [inline lib/ansible/plugins/become/pmrun.py:65]: This is very concern to us regarding the subsequent adoption by AWX / Tower. We need these prompts to be machine-parsable. While, yes, you do provide documentation that will probably have the prompt text in `ansible-doc`, correctly using that to get this info assumes a great deal of things are connected which are very specific to the specific project folder structure, Ansible config, and other things.

You can see our implementation here:

https://github.com/ansible/awx/blob/devel/awx/main/tasks.py#L1354-L1357

We are using the pexpect library to search for the string "PMRUN password:" in the standard out. If that pattern is broken, I don't know how we will deal with it.

Comment (sivel) [reply lib/ansible/plugins/become/pmrun.py:65]: I believe this is the prompt that we expect `pmrun` to prompt us with, not what we will prompt the user with:

https://github.com/ansible/ansible/pull/50421/files#diff-6ce98545d6c25c12c5747868c8866e6aR61

Comment (jbradberry) [reply lib/ansible/plugins/become/pmrun.py:65]: Tagging myself in since I'll be working on the AWX support...

Comment (AlanCoding) [reply lib/ansible/plugins/become/pmrun.py:65]: Ah, I see, that's great if that's the case. I think we just need to test this out.

**Pull request closed without merging**